Code/src/main/java/nl/utwente/group10/haskell/catalog/HaskellClassCatalog.java

```diff
@@ -0,0 +1,132 @@
+package nl.utwente.group10.haskell.catalog;
+
+import nl.utwente.group10.haskell.env.Env;
+import nl.utwente.group10.haskell.exceptions.CatalogException;
+import nl.utwente.group10.haskell.type.TypeClass;
+import org.w3c.dom.Document;
+import org.w3c.dom.NamedNodeMap;
+import org.w3c.dom.Node;
+import org.w3c.dom.NodeList;
+import org.xml.sax.SAXException;
+
+import javax.xml.parsers.DocumentBuilder;
+import javax.xml.parsers.DocumentBuilderFactory;
+import javax.xml.parsers.ParserConfigurationException;
+import javax.xml.validation.SchemaFactory;
+import java.io.IOException;
+import java.net.URL;
+import java.util.*;
+
+/**
+ * Provides a convenient interface to the XML catalog of available Haskell classes.
+ */
+public class HaskellClassCatalog {
```

Comment:
I'm not sure how I feel about splitting the Catalog in two - it seems you'd have to duplicate quite some XML-related nonsense in doing so. A possible alternative would be to have one catalog that knows about both functions and typeclasses, a `catalog.xml` if you will:

````
<catalog>
  <functions>
    <!-- as before -->
  </functions>
  <typeclasses>
    <!-- as here -->
  </typeclasses>
</catalog>
````